log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/RollingFileManager.java

```diff
@@ -636,7 +643,7 @@ private boolean rollover(final RolloverStrategy strategy) {
                     asyncExecutor.execute(new AsyncAction(descriptor.getAsynchronous(), this));
                     releaseRequired = false;
                 }
-                return success;
+                return true;
```

Comment:
There are a lot of `boolean`s in this method with different purposes. It might be useful to rename them to make the code more clear. For example:

- `releaseRequired` actually tells us if the asynchronous action has started. Maybe we could rename it to `asyncActionStarted` and define it only where it is needed (just after the first `try`).
- `success` tells use if the synchronous action has succeeded. It should be rather called something like `syncActionSuccess` and be defined just before the synchronous action is executed.
- A third unnamed `boolean` is used in the return statements and there is no comment stating what `rollover(RolloverStrategy)` returns. I think we should introduce a `outputStreamClosed` variable that is defined at the beginning of the method and used in all the return statements. The lack of documentation is what caused #2592 in the first place.